scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,100 @@
+# IPC gullwing generator
+
+This generator uses IPC-7351B equations and by default fillet definitions to derive a footprint for gullwing stile packages.
+Examples of supported packages include: QFN, SOIC, SO
+
+## Running the script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_gullwing_generator.py size_definitions/eqfp.yaml` (replace eqfp.yaml with the size definition file that contains your part.)
+
+## Size definition format
+
+Every file contains a header with parameters applied to all parts. These define the common footprint name prefix and the output library.
+
+```yaml
+FileHeader:
+  library_Suffix: 'QFP' #resulting library name Package_QFP
+  device_type: 'EQFP' #footprint names will be of style ...EQFP-pincount_...
+```
+
+---
+
+Every further entry in the script is assumed to be a package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was genererated by which parameter set.)
+
+``` yaml
+internal_name:
+  # parameter list
+```
+
+The following parameters are supported for every package
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- footprint name control
+  - Manufacturer if specific to one manufacturer (`manufacturer`) {string}
+  - Part number if specific to one part (`part_number`) {string}
+  - Custom naming (`custom_name_format`) {format string}
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Over all size representing lead tip to lead tip dimension (`overall_size_x`, `overall_size_y`) {dimension}
+  - Lead with (`lead_width`) {dimension}
+  - Lead length representing tip till bend (`lead_len`)
```

Comment:
'till' has an extra 'l'. As there are two lead bends perhaps a better wording would be 'distance lead is roughly parallel to the PCB, from the lead bend closest to the PCB surface to the lead tip'?

Reply:
The measurement in question is the one projected to the pcb plane. Meaning the bend angle tolerance is already part of its tolerance.

I think i will include a picture in the final version.

Reply:
The typo with the extra 'l' was not fixed.

Reply:
What typo do you mean exactly?

Reply:
The general informality of the statement; perhaps change `lead_len` description to `Distance between the lead bend and lead tip`?

Reply:
Also, this line does not end in the data type in curved brackets.

Reply:
fixed